docs/pages/server-access/guides/ec2-discovery.mdx

```diff
@@ -0,0 +1,277 @@
+---
+title: Configure Teleport to Automatically Enroll EC2 instances
+description: How to configure Teleport to automatically enroll EC2 instances.
+---
+
+Teleport can be configured to run a discovery service that will
+connect to EC2 and automatically discover and enroll EC2 instances
+matching configured labels. It will then execute an install script on
+these discovered instances using AWS Systems Manager, that will install
+Teleport, start it and join the cluster.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- AWS account with EC2 instances and permissions to create and attach IAM policies as well as permission to create Systems Manager documents.
+- EC2 instances running Ubuntu/Debian/RHEL/Amazon Linux 2 if making use of the default Teleport install script.
+- A running Teleport Cluster.
+
+## Step 1/5. Create an EC2 invite token
+
+When discovering EC2 nodes, Teleport makes use of IAM invite tokens for
+authenticating joining Nodes.
+
+Create the file `token.yaml`:
+```yaml
+# token.yaml
+---
+kind: token
+version: v2
+metadata:
+  # the token name is not a secret because instances must prove that they are
+  # running in your AWS account to use this token
+  name: aws-discovery-iam-token
+  # set a long expiry time, the default for tokens is only 30 minutes
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  # use the minimal set of roles required
+  roles: [Node]
+
+  # set the join method allowed for this token
+  join_method: iam
+
+  allow:
+  # specify the AWS account which nodes may join from
+  - aws_account: "123456789"
+```
+
+Add the token to the Teleport cluster with:
+```
+tctl create -f token.yaml
+```
+
+More documentation on IAM invite tokens can be found [here](../../management/guides/joining-nodes-aws-iam.mdx)
```

Comment:
We should also move this link to a "Next steps" section, since it's not required for following this guide. We want to get users to the end of the guide as quickly as possible.